app/scripts/modules/core/src/slack/SlackReader.spec.ts

```diff
@@ -24,15 +24,10 @@ const mockSlackChannels: ISlackChannel[] = [
 
 describe('SlackReader', () => {
   it('should fetch a list of public Slack channels', async () => {
-    spyOn(SlackReader, 'getChannels').and.callThrough();
-    spyOn(API, 'one')
-      .and.callThrough()
-      .and.returnValue({ getList: () => Promise.resolve(mockSlackChannels) });
+    const http = mockHttpClient({ autoFlush: true });
+    http.expectGET('/slack/channels').respond(200, mockSlackChannels);
 
-    await SlackReader.getChannels().then((channels: ISlackChannel[]) => {
-      expect(SlackReader.getChannels).toHaveBeenCalled();
-      expect(channels.length).toEqual(2);
-      expect(API.one).toHaveBeenCalledWith('slack', 'channels');
-    });
+    const channels = await SlackReader.getChannels();
+    expect(channels.length).toEqual(2);
   });
 });
```

Comment:
I quite like the improvements made to the developer experience as a result of this refactor - much more intuitive interaction, as a layperson.